scripts/install.sh

```diff
@@ -1,159 +1,290 @@
+  debug "$1 installation complete."
+}
+
+get_photonvision_releases() {
+  if ! command -v curl > /dev/null 2>&1 ; then
+    die "./install --list-versions requires curl and it is not installed."
+  fi
+
+  if [ -z "$PHOTON_VISION_RELEASES" ] ; then
+    PHOTON_VISION_RELEASES="$(curl -sk https://api.github.com/repos/photonvision/photonvision/releases)"
+  fi
+
+ echo "$PHOTON_VISION_RELEASES"
+}
+
+get_versions() {
+  if [ -z "$PHOTON_VISION_VERSIONS" ] ; then
+    PHOTON_VISION_VERSIONS=$(get_photonvision_releases | \
+      sed -En 's/\"tag_name\": \"v([0-9]+\.[0-9]+\.[0-9]+)(-(beta|alpha)(-[0-9])?(\.[0-9]+)?)?\",/\1\2/p' | \
+      sed 's/^[[:space:]]*//')
+  fi
+
+  echo "$PHOTON_VISION_VERSIONS"
+}
+
+is_version_available() {
+  local target_version="$1"
+
+  # latest is a special case
+  if [ "$target_version" = "latest" ]; then
+    return
+  fi
+
+  # Check if the version is present
+  if ! echo "$(get_versions)" | grep -qFx "$target_version"; then
+    return 1
+  fi
+
+  # Check if multiple lines are match. You can only match 1.
+  local line_count="$(echo "$versions" | grep -cFx "$target_string")"
+  if [ $line_count -gt 1 ] ; then
+    return 1
+  fi
+
+  return 0
+}
+
 help() {
-  echo "This script installs Photonvision."
-  echo "It must be run as root."
-  echo
-  echo "Syntax: sudo ./install.sh [-h|m|n|q]"
-  echo "  options:"
-  echo "  -h        Display this help message."
-  echo "  -m        Install and configure NetworkManager (Ubuntu only)."
-  echo "  -n        Disable networking. This will also prevent installation of NetworkManager."
-  echo "  -q        Silent install, automatically accepts all defaults. For non-interactive use."
-  echo
+  cat << EOF
+This script installs Photonvision.
+It must be run as root.
+
+Syntax: sudo ./install.sh [options]
+  options:
+  -h, --help
+      Display this help message.
+  -l, --list-versions
+      Lists all available versions of PhotonVision.
+  -v <version>, --version=<version>
+      Specifies which version of PhotonVision to install.
+      If not specified, the latest stable release is installed.
+      Ignores leading 'v's.
+  -a <arch>, --arch=<arch>
+      Install PhotonVision for the specified architecture.
+      Supported values: aarch64, x86_64
+  -m [option], --install-nm=[option]
+      Whether or not to install NetworkManager. Only used on
+      Ubuntu, and ignored for all other distros.
+      Supported options are: "yes", "no", and "ask".
+      "ask" prompts the user for installation of NetworkManager.
+      If not specified, will fall back to "ask".
+      If specified, "yes" is the default option.
+  -n, --no-networking
+      Disable networking. This will also prevent installation of
+      NetworkManager (ignoring -m,--install-nm).
+  -q, --quiet
+      Silent install, automatically accepts all defaults. For
+      non-interactive use. Forces --install-nm="no".
+
+EOF
 }
 
-INSTALL_NETWORK_MANAGER="false"
+INSTALL_NETWORK_MANAGER="ask"
 
```

Comment:
Oof, yeah. I think I missed that. I'll patch that really quick. I've mostly tested it on my computer. I can test it on a pi running raspbian sometime in the next week.

Reply:
Actually wait can you clarify what you mean by "missing setting the default version"? The default version is the latest release.

Reply:
Oh wait never mind. I know what you mean. I need to edit this so it doesn't require the version. That should be a default option like the way `-m,--install-nm`

Reply:
Will that make it the default even when the option isn’t passed at all?

Reply:
I believe the current behavior after the patch is that if no option is passed, latest is installed. If an `-v` is specified, an argument is required. You can enter a version with either a leading v or not and it will check using the GitHub API for releases using curl (the check happens after curl is installed by the script). Currently `./install -l` lists all versions but fails if you don't have `curl`. I think I'll add a commit that prompts the user if they want to install it if it's not present to reduce friction.

Reply:
I believe that `wget -qO- URL` will do the same thing as `curl -sk URL`. If they followed our instructions, they used `wget` to download the script, so they should already have it.

Reply:
It's too bad that we can't assume `jq` is installed. Then your `sed` commands could be replaced with `| jq -r '.[].tag_name | select(. != "Dev") | ltrimstr("v")'`.